src/wp-includes/blocks.php

```diff
@@ -148,48 +148,74 @@ function register_block_script_handle( $metadata, $field_name ) {
  *
  * @param array  $metadata   Block metadata.
  * @param string $field_name Field name to pick from metadata.
- * @return string|false Style handle provided directly or created through
+ * @return array|string|false Style handle provided directly or created through
  *                      style's registration, or false on failure.
  */
+
 function register_block_style_handle( $metadata, $field_name ) {
 	if ( empty( $metadata[ $field_name ] ) ) {
 		return false;
 	}
+	$style_data = $metadata[ $field_name ];
+	if ( ! is_array( $style_data ) ) {
+		return _register_single_block_style_handle( $metadata, $field_name, $style_data );
+	}
+
+	/*
+	 * $style_data is a list of stylesheet paths and
+	 * style arrays such as { "color": { "text": "#fff" } }.
+	 *
+	 * In here, we replace the stylesheet paths with registered style handles.
+	 */
+	$handles = array();
+	foreach ( $style_data as $style_item ) {
+		if ( is_array( $style_item ) ) {
+			$handles[] = $style_item;
+		} else {
+			$handles[] = _register_single_block_style_handle( $metadata, $field_name, $style_item );
+		}
+	}
+
+	return $handles;
+}
+
+function _register_single_block_style_handle( $metadata, $field_name, $style_data ) {
```

Comment:
This probably needs a better name.

Reply:
Agreed! Any suggestions?

Reply:
What about `register_single_style_handle_for_block`? I don't think we need the `_` prefix.